File: bjtu_oj/CPP/1864.cpp
#include <stdio.h>
#include <math.h>
#include <vector>
using namespace std;

int DigitSum(int n);
bool IsPrime(int n);

int main()
{
    int n, k;
    scanf("%d %d", &n, &k);
    vector<int> arr;
    for (int i = k; i <= n; i++)
    {
        if (DigitSum(i) == k)
        {
            arr.push_back(i);
        }
    }
    int res = 0;
    for (int v : arr)
    {
        if (IsPrime(v))
        {
            res++;
        }
    }
    printf("%d\n", res);
    return 0;
}

int DigitSum(int n)
{
    int res = 0;
    int t = n;
    while (t >= 1)
    {
        res += t % 10;
        t /= 10;
    }
    return res;
}

bool IsPrime(int n)
{
    for (int i = 2; i <= sqrt(n); i++)
    {
        if (n % i == 0)
        {
            return false;
        }
    }
    return true;
}

File: bjtu_oj/1864.go
package main

import (
	"fmt"
	"math"
)

func main()  {
	var n, k int
	fmt.Scanf("%d %d", &n, &k)
	var arr []int
	for i := k; i <= n; i++ {
		if DigitSum(i) == k {
			arr = append(arr, i)
		}
	}
	var res = 0
	for _, v := range arr {
		if IsPrime(v) {
			res++
		}
	}
	fmt.Println(res)
}

func DigitSum(n int) int {
	var res = 0
	t := n
	for t >= 1{
		res += t%10
		t /= 10
	}
	return res
}

func IsPrime(n int) bool {
	for i := 2; float64(i) <= math.Sqrt(float64(n)); i++ {
		if n%i == 0 {
			return false
		}
	}
	return true
}

